### PR TITLE
Improve partial parsing docs

### PR DESCRIPTION
**Pull request by tatiana**
Improves docs to highlight the limitation of the parsing parsing approach (introduced in #800), following up the feedback on #722

### CODE REVIEW

**Comment by tatiana** [comment]
@dwreeves if you have a chance, I'd love your feedback on this

File: docs/configuration/project-config.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ variables that should be used for rendering and execution. It takes the followin
   env vars is only supported when using ``RenderConfig.LoadMode.DBT_LS`` load mode.
 - ``partial_parse``: (new in v1.4) If True, then attempt to use the ``partial_parse.msgpack`` if it exists. This is only used
   for the ``LoadMode.DBT_LS`` load mode, and for the ``ExecutionMode.LOCAL`` and ``ExecutionMode.VIRTUALENV``
-  execution modes.
+  execution modes. Due to the way that dbt `partial parsing works <https://docs.getdbt.com/reference/parsing#known-limitations>`_, it does not work with Cosmos profile mapping classes. To benefit from this feature, users have to set the ``profiles_yml_filepath`` argument in ``ProfileConfig``.
```
Comment (dwreeves) [inline docs/configuration/project-config.rst:28]: The exception to this is that you can copy+paste the YAML generated by Cosmos, but I feel like that's splitting hairs and probably isn't worth mentioning.

**Comment by dwreeves** [comment]
One thing that is probably worth mentioning: Partial parsing requires the same `--vars`, `--target`, `--profile`, and env vars (as called by the `env_var()` macro) to be equal in each run, otherwise it doesn't work. [This specifically includes `profiles.yml` env vars](https://github.com/dbt-labs/dbt-core/blob/c6c0c79216a3c8032d5ea5ed608ca2ef7bdacc9a/core/dbt/parser/manifest.py#L838C1-L844C65).

For practical purposes, this means that the connection that generates the `manifest.json` must also be the same that actually utilizes it. So if you have for example a `CICD_USER` which runs `dbt parse` and an` AIRFLOW_USER` that actually executes in Cosmos, this won't work.

This also may end up meaning that users relying on `--vars` may end up running into issues (I have an idea for `1.5.0` to work around that a bit...).

**Review by pankajastro — APPROVED** [review]
LGTM

**Comment by tatiana** [comment]
@dwreeves, thanks for the detailed review; I tried to update with your feedback! Please let me know if I missed anything.
@pankajastro I made a few changes based on PR #800. Could you confirm that you're still happy with it?

**Comment by dwreeves** [comment]
Looks good but I've been out of the loop for the last month or two, and didn't realize you worked on a solution to make partial parsing better. 👀 Very nice!!! I need to catch up.

File: docs/configuration/partial-parsing.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Profile configuration
+---------------------
+
+To respect the dbt requirement of having the same profile to benefit from partial parsing, Cosmos users should either:
+* If using Cosmos profile mapping (``ProfileConfig(profile_mapping=...``), disable using mocked profile mappings by setting ``render_config=RenderConfig(enable_mock_profile=False)``
+* Declare their own ``profiles.yml`` file, via ``ProfileConfig(profiles_yml_filepath=...)``
+
+If users don't follow these guidelines, Cosmos will use different profiles to parse the dbt project and to run tasks, and the user won't leverage dbt partial parsing.
+Their logs will contain multiple ``INFO`` messages similar to the following, meaning that Cosmos are is not using partial parsing:
```
[the remark this answers is not in the document]
Comment (tatiana) [reply docs/configuration/partial-parsing.rst]: Addressed in 081beccd1f25de45529972067d1e1a7d72d6fc73

File: docs/configuration/partial-parsing.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+If the dbt project ``target`` directory has a ``partial_parse.msgpack``, Cosmos will attempt to use it.
+
+There is a chance, however, that the file is stale or was generated in a way that is different to how Cosmos runs the dbt commands.
+
+Therefore, Cosmos also caches the most up-to-date ``partial_parse.msgpack`` file after running a dbt command in the `system temporary directory <https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir>`_.
+With this, unless there are code changes, each Airflow node should only run the dbt command with a full dbt project parse once, and benefit from partial parsing from then onwards.
+
+It is possible to override the directory that Cosmos uses caching with the Airflow configuration ``[cosmos][cache_dir]`` or environment variable ``AIRFLOW__COSMOS__CACHE_DIR``.
+
+To turn off caching, set the Airflow configuration ``[cosmos][enable_cache]`` or the environment variable ``AIRFLOW__COSMOS__ENABLE_CACHE=0``.
```
[the remark this answers is not in the document]
Comment (tatiana) [reply docs/configuration/partial-parsing.rst]: Thanks, @pankajkoti , addressed in 081beccd1f25de45529972067d1e1a7d72d6fc73

**Comment by tatiana** [comment]
Since this relies on the changes of #904, and once this PR is merged, it will be published into Cosmos official docs, I'll wait for #904 to be merged beforehand

**Pull request merged**